### PR TITLE
Revert "fix: Also build rdkafka with ssl if ssl feature is enabled"

### PR DESCRIPTION
**Pull request by untitaker**
Reverts getsentry/relay#888
Reverts getsentry/relay#881

We should find a different solution that does not prevent us from using `--all-features`.

#skip-changelog

### CODE REVIEW

**Review by jan-auer — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -14,7 +14,6 @@
 
 **Features**:
 
-- Add `kafka-ssl` compilation feature that builds Kafka linked against OpenSSL. This feature is enabled in Docker containers only. This is only relevant for Relays running as part of on-premise Sentry. ([#881](https://github.com/getsentry/relay/pull/881))
```
Comment (jan-auer) [inline CHANGELOG.md]: Since this is already released, we need to create a new changelog entry.

**Review by jan-auer — CHANGES_REQUESTED** [review]

**Comment by untitaker** [comment]
@ajacques this PR reverts https://github.com/getsentry/relay/pull/881 as the feature-flagging setup broke our development workflow

I suggest maintaining a fork without this revert, or doing something to make `--all-features` usable even with kafka-ssl feature on OS X. We cannot ask people to install OpenSSL on OS X.

**Review by jan-auer — APPROVED** [review]

**Comment by ajacques** [comment]
> @ajacques this PR reverts #881 as the feature-flagging setup broke our development workflow
> 
> I suggest maintaining a fork without this revert, or doing something to make `--all-features` usable even with kafka-ssl feature on OS X. We cannot ask people to install OpenSSL on OS X.

Why is this a new breaking change? It was my understanding that Relay supported SSL on the HTTP side.

**Comment by jan-auer** [comment]
@ajacques Unfortunately, these are two completely separate concerns. You're right, Relay does support incoming SSL connections and while optional we build this by default. It also supports communicating with other Relays via HTTPS.

However, this configuration of Relay ("processing mode") is designed to run within the same network as the Sentry Kafka broker. For this reason, we've never had to support any SSL, since access to that broker can and should be guarded. In fact, these are completely separate functionalities and for that reason also controlled separately. 

There are several factors that in combination make it not straightforward to add this feature flag. Since it was blocking our development workflow, we had to back it out.

Could you highlight why you are operating your Sentry Kafka broker with SSL?

**Comment by ajacques** [comment]
Maybe I'm missing something, but it's not clear why if you support OpenSSL already to terminate SSL or connect to upstream, that same flag can't also be used to enable/disable for the rdkafka. e.g. If you already use OpenSSL to terminate/connect upstream, how does that work on Mac today? I apologize if I'm just dense here. I'll take another look at the code. Maybe I can come up with a solution.

I'm trying to deploy in an environment where we avoid trusting the underlying network and use encryption/credentials for every hop. Even if it's on the same VPC and using network-level access controls. I can request approval to run it without this, but it'd make it easier with encryption/auth at every level.

I appreciate your attempt at fixing this issue.

**Comment by untitaker** [comment]
>If you already use OpenSSL to terminate/connect upstream, how does that work on Mac today?

on mac we don't depend on openssl in any scenario. We use https://crates.io/crates/native-tls which dispatches to openssl only on Linux. Kafka doesn't do this, it depends on OpenSSL in all scenarios. That's why kafka-ssl was its own featureflag in the first place: Because it pulls in OpenSSL as dependency outside of Linux for the first time.

**Pull request merged**